From: Ben Hutchings <ben@decadent.org.uk>
Subject: tools/hv: Fix /var subdirectory

We will install this in /usr, so it must use /var/lib for its state.
Only programs installed under /opt should use /var/opt.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
--- a/tools/hv/hv_kvp_daemon.8
+++ b/tools/hv/hv_kvp_daemon.8
@@ -18,7 +18,7 @@
 .SH FILES
 .ta
 .nf
-/var/opt/hyperv/.kvp_pool_*
+/var/lib/hyperv/.kvp_pool_*
 .fi
 
 .SH AUTHORS
--- a/tools/hv/hv_kvp_daemon.c
+++ b/tools/hv/hv_kvp_daemon.c
@@ -216,9 +216,9 @@
 	int i;
 	int alloc_unit = sizeof(struct kvp_record) * ENTRIES_PER_BLOCK;
 
-	if (access("/var/opt/hyperv", F_OK)) {
-		if (mkdir("/var/opt/hyperv", S_IRUSR | S_IWUSR | S_IROTH)) {
-			syslog(LOG_ERR, " Failed to create /var/opt/hyperv");
+	if (access("/var/lib/hyperv", F_OK)) {
+		if (mkdir("/var/lib/hyperv", S_IRUSR | S_IWUSR | S_IROTH)) {
+			syslog(LOG_ERR, " Failed to create /var/lib/hyperv");
 			exit(EXIT_FAILURE);
 		}
 	}
@@ -227,7 +227,7 @@
 		fname = kvp_file_info[i].fname;
 		records_read = 0;
 		num_blocks = 1;
-		sprintf(fname, "/var/opt/hyperv/.kvp_pool_%d", i);
+		sprintf(fname, "/var/lib/hyperv/.kvp_pool_%d", i);
 		fd = open(fname, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR | S_IROTH);
 
 		if (fd == -1)
